### PR TITLE
Update CHANGELOG and prepare for release

### PR DESCRIPTION
**Pull request by cpiemontese**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,41 +3,57 @@
...
+[Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-### [Next]
+## [Unreleased]
+
+## [0.4.2]
+
+### Changed
+
+- update itertools to 0.13.0
```
Comment (23Skidoo) [inline CHANGELOG.md]: Also `positional_derive`'s dependency on `syn` seems to have been bumped to v2.

**Review by 23Skidoo — DISMISSED** [review]
LGTM. I also checked that `accielle` builds against this branch.

File: CHANGELOG.md (hunk 1 above)
Comment (cpiemontese) [reply CHANGELOG.md]: I felt there had been other changes, thanks!

**Comment by cpiemontese** [comment]
@23Skidoo we know next to nothing about this library, do you think it would make sense for your team or domain to own it...?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,41 +3,59 @@
...
+[Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-### [Next]
+## [Unreleased]
+
+## [0.4.2]
+
+### Changed
+
+- update criterion to 0.5
```
Comment (23Skidoo) [inline CHANGELOG.md:15]: criterion is a dev dependency, but ok

**Review by 23Skidoo — APPROVED** [review]
LGTM

**Comment by 23Skidoo** [comment]
@cpiemontese I'll raise it with my team.

**Pull request merged**